kotlinx-coroutines-test/common/src/TestCoroutineScheduler.kt

```diff
@@ -193,6 +193,18 @@ public class TestCoroutineScheduler : AbstractCoroutineContextElement(TestCorout
      * Consumes the knowledge that a dispatch event happened recently.
      */
     internal val onDispatchEvent: SelectClause1<Unit> get() = dispatchEvents.onReceive
+
+    /**
+     * Returns a [TimeSource] synced with this [currentTime] to allow time based consumers handling this delay-skipping
+     * behavior.
+     */
+    @ExperimentalTime
+    public val timeSource: TimeSource = VirtualTimeSource()
+
+    @ExperimentalTime
+    private inner class VirtualTimeSource: AbstractLongTimeSource(DurationUnit.MILLISECONDS) {
```

Comment:
simply changing it to an object does not work, because the class and the object are unrelated, so it is not possible to access `currentTime` (and inner objects are not available). An anonymous class/object in the getter does work, but would allocate a new instance every call, would it?

Reply:
Oh yeah, that's true :D Changed